core-bundle/src/Monolog/ContaoTableHandler.php

```diff
@@ -65,22 +63,22 @@ public function handle(array $record): bool
 
     protected function write(array $record): void
     {
-        $this->createStatement();
-
         /** @var \DateTime $date */
         $date = $record['datetime'];
 
         /** @var ContaoContext $context */
         $context = $record['extra']['contao'];
 
-        $this->statement->executeStatement([
+        $this->getConnection()->insert('tl_log', [
             'tstamp' => $date->format('U'),
             'text' => StringUtil::specialchars((string) $record['formatted']),
             'source' => (string) $context->getSource(),
             'action' => (string) $context->getAction(),
             'username' => (string) $context->getUsername(),
             'func' => $context->getFunc(),
             'browser' => StringUtil::specialchars((string) $context->getBrowser()),
+            'uri' => StringUtil::specialchars(urldecode($context->getUri() ?? '')),
```

Comment:
```suggestion
            'uri' => StringUtil::specialchars($context->getUri() ?? ''),
```

I don’t think it should be URL-decoded here (and this would potentially be dangerous)